crates/ruff_linter/src/rules/flake8_use_pathlib/rules/replaceable_by_pathlib.rs

```diff
@@ -31,7 +31,35 @@ pub(crate) fn replaceable_by_pathlib(checker: &Checker, call: &ExprCall) {
             // PTH103
             ["os", "mkdir"] => Some(OsMkdir.into()),
             // PTH104
-            ["os", "rename"] => Some(OsRename.into()),
+            ["os", "rename"] => {
+                // `src_dir_fd` and `dst_dir_fd` are not supported by pathlib, so check if they are
+                // are set to non-default values.
+                // Signature as of Python 3.13 (https://docs.python.org/3/library/os.html#os.rename)
+                // ```text
+                //           0    1       2                3
+                // os.rename(src, dst, *, src_dir_fd=None, dst_dir_fd=None)
+                // ```
+                if call
+                    .arguments
+                    .find_argument_value("src_dir_fd", 2)
+                    .is_some_and(|expr| !expr.is_none_literal_expr())
```

Comment:
Can we add a test where the `src_dir_fd` is specified by position (and not name) and another test where `src_dir_fd` is `None`. The same for `dst_dir_fd`